Apps/Examples/Examples/All Examples/NavigationSimulatorExample.swift

```diff
@@ -0,0 +1,211 @@
+            try mapView.mapboxMap.style.addPersistentLayer(makeCasingLayer())
+            try mapView.mapboxMap.style.addPersistentLayer(makeRouteLineLayer())
+
+            navigationSimulator.start()
+        } catch {
+            print("Unexpected error when adding source/style: \(error)")
+        }
+    }
+
+    // MARK: - Util
+
+    private func makeRouteLineLayer() -> LineLayer {
+        var routeLayer = LineLayer(id: ID.routeLineLayer)
+        routeLayer.source = ID.routeSource
+        routeLayer.lineCap = .constant(.round)
+        routeLayer.lineJoin = .constant(.round)
+        routeLayer.lineWidth = .expression(
+            Exp(.interpolate) {
+                Exp(.exponential) {
+                    1.5
+                }
+                Exp(.zoom)
+                4.0
+                Exp(.product) {
+                    3.0
+                    1.0
+                }
+                10.0
+                Exp(.product) {
+                    4.0
+                    1.0
+                }
+                13.0
+                Exp(.product) {
+                    6.0
+                    1.0
+                }
+                16.0
+                Exp(.product) {
+                    10.0
+                    1.0
+                }
+                19.0
+                Exp(.product) {
+                    14.0
+                    1.0
+                }
+                22.0
+                Exp(.product) {
+                    18.0
+                    1.0
+                }
+            }
+        )
+        routeLayer.lineGradient = .expression(
+            Exp(.interpolate) {
+                Exp(.linear)
+                Exp(.lineProgress)
+                0.0
+                UIColor(red: 6.0/255.0, green: 1.0/255.0, blue: 255.0/255.0, alpha: 1)
+                0.1
+                UIColor(red: 59.0/255.0, green: 118.0/255.0, blue: 227.0/255.0, alpha: 1)
+                0.3
+                UIColor(red: 7.0/255.0, green: 238.0/255.0, blue: 251.0/255.0, alpha: 1)
+                0.5
+                UIColor(red: 0, green: 255.0/255.0, blue: 42.0/255.0, alpha: 1)
+                0.7
+                UIColor(red: 255.0/255.0, green: 252.0/255.0, blue: 0, alpha: 1)
+                1.0
+                UIColor(red: 255.0/255.0, green: 30.0/255.0, blue: 0, alpha: 1)
+            }
+        )
+
+        return routeLayer
+    }
+
+    private func makeCasingLayer() -> LineLayer {
+        var casingLayer = LineLayer(id: ID.casingLineLayer)
+        casingLayer.source = ID.routeSource
+        casingLayer.lineCap = .constant(.round)
+        casingLayer.lineJoin = .constant(.round)
+
+        casingLayer.lineWidth = .expression(
+            Exp(.interpolate) {
+                Exp(.exponential) {
+                    1.5
+                }
+                Exp(.zoom)
+                10.0
+                Exp(.product) {
+                    7.0
+                    1.0
+                }
```

Comment:
I'm not quite sure if I understand this correctly, but shouldn't it translate to something like `7.0 * 1.0` - given that `.product` - "Returns the product of the inputs"?

Reply:
yes, I think explicitly putting one here is for the sake of readiblity in its string representation :-D we could instead do `Exp(.literal) { value }` but I think we should keep this to make it consistent with Android.

Reply:
I too think the syntax of iOS `Expression` could be improved. At the moment it can take up too many lines of code and provide little readability :-D would be nice to have a team discussion on this and propose new or addition way for Expression.